File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
    int answer = 0;
    stack<int> s;
    for (const auto& num : moves) {
        for (int i = 0; i < board.size(); i++) {
            if (board[i][num - 1] != 0) {
                int doll = board[i][num - 1];
                board[i][num - 1] = 0;
                
                if (!s.empty() && s.top() == doll) {
                    s.pop();
                    answer += 2;
                }
                else {
                    s.push(doll);
                }
                
                break;
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    temp = []
    
    for i in moves:
        for j in range(len(board)):
            if board[j][i - 1] != 0:
                doll = board[j][i - 1]
                board[j][i - 1] = 0
                
                if temp and temp[-1] == doll:
                    temp.pop()
                    answer += 2
                else:
                    temp.append(doll)
                    
                break
                
    return answer